internal/controller/datamovement_controller.go

```diff
@@ -513,13 +501,317 @@ func buildDMCommand(ctx context.Context, profile dmConfigProfile, hosts []string
 		}
 	}
 
-	return strings.Split(cmd, " "), hostfile, nil
+	return strings.Split(cmd, " "), nil
+}
+
+func (r *DataMovementReconciler) prepareDestination(ctx context.Context, dm *nnfv1alpha1.NnfDataMovement, mpiHostfile string, log logr.Logger) error {
+	// These functions interact with the filesystem, so they can't run in the test env
+	if !isTestEnv() {
+		// Determine the destination directory based on the source and path
+		log.Info("Determining destination directory based on source/dest file types")
+		destDir, err := getDestinationDir(dm, mpiHostfile, log)
```

Comment:
prepareDestination() is the only caller of getDestination() and handleIndexMountDir(). Both of those functions call isSourceAFile() and isDestAFile(), and both of _those_ functions use mpiIsDir(). This feels like repeat calls, with the same args, to mpirun.